File: BOEC-finally-war/web/AdminCssJs/js/addElectronic.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

$('#eType').on('change', function () {
    resetAttr();
    var eType = $('#eType').val();
    if (eType === 'SmartPhone' || eType === 'Tablet') {
        $('#eBattery').attr('disabled', false);
        $('#eCameraAfter').attr('disabled', false);
        $('#eIsDVD').attr('disabled', true);
        $('#eIsKeyBoardLight').attr('disabled', true);
    } else if (eType === 'Laptop') {
        $('#eBattery').attr('disabled', true);
        $('#eCameraAfter').attr('disabled', true);
        $('#eIsDVD').attr('disabled', false);
        $('#eIsKeyBoardLight').attr('disabled', false);
    }
});
function resetAttr() {
    $('#eBattery').val('');
    $('#eCameraAfter').val('');
    $('#eIsDVD').val('');
    $('#eIsKeyBoardLight').val('');
}
function readURL(input) {
//                debugger;
    console.log(input.files);
    console.log(input.files[0]);
    if (input.files && input.files[0]) {
        var reader = new FileReader();

        reader.onload = function (e) {
            $('#image')
                    .attr('src', e.target.result)
                    .width(50)
                    .height(50);
        };
        reader.readAsDataURL(input.files[0]);
    }
}

$('#btnThem').off('click').on('click', function () {
    var eType = $('#eType').val();
    var eName = $('#eName').val();
    var ePrice = $('#ePrice').val();
    var eQuantity = $('#eQuantity').val();
//    var eImage = $('#eImage').val();
//    var image = $('#image').val();
    var eCompanyID = $('#eCompanyID').val();
    var eScreenSize = $('#eScreenSize').val();
    var eOperaSystem = $('#eOperaSystem').val();
    var eWarrantyPeriod = $('#eWarrantyPeriod').val();
    var eBattery = $('#eBattery').val();
    var eCameraAfter = $('#eCameraAfter').val();
    var eIsDVD = $('#eIsDVD').val();
    var eIsKeyBoardLight = $('#eIsKeyBoardLight').val();
    if (eType == '' || eType == 'undefined' || eType == null) {
        alert("Bạn chưa chọn kiểu Electronic");
        $('#eType').focus();
        return;
    }
    if (eName == '' || eName == 'undefined' || eName == null) {
        alert("Bạn chưa tên Electronic");
        $('#eName').focus();
        return;
    }
    if (ePrice == '' || ePrice == 'undefined' || ePrice == null) {
        alert("Bạn chưa nhập giá");
        $('#ePrice').focus();
        return;
    }

//    if (eImage == '' || eImage == 'undefined' || eImage == null) {
//        alert("Bạn chưa chọn hình ảnh");
//        $('#eImage').focus();
//        return;
//    }
    if (eQuantity == '' || eQuantity == 'undefined' || eQuantity == null) {
        alert("Bạn chưa nhập số lượng");
        $('#eQuantity').focus();
        return;
    }
    if (eCompanyID == '' || eCompanyID == 'undefined' || eCompanyID == null) {
        alert("Vui lòng chọn hãng sản xuất");
        $('#eCompanyID').focus();
        return;
    }
    if (eScreenSize == '' || eScreenSize == 'undefined' || eScreenSize == null) {
        alert("Vui lòng chọn kích thước màn hình");
        $('#eScreenSize').focus();
        return;
    }
    if (eOperaSystem == '' || eOperaSystem == 'undefined' || eOperaSystem == null) {
        alert("Vui lòng nhập hệ điều hành");
        $('#eOperaSystem').focus();
        return;
    }
    if (eWarrantyPeriod == '' || eWarrantyPeriod == 'undefined' || eWarrantyPeriod == null) {
        alert("Vui lòng nhập số năm bảo hành");
        $('#eWarrantyPeriod').focus();
        return;
    }

    if (eType === 'SmartPhone' || eType === 'Tablet') {     // Kiểm tra Type Electronic
        if (eBattery == '' || eBattery == 'undefined' || eBattery == null) {
            alert("Bạn chưa nhập dung lượng pin");
            $('#eBattery').focus();
            return;
        }
        if (eCameraAfter == '' || eCameraAfter == 'undefined' || eCameraAfter == null) {
            alert("Bạn chưa nhập số chấm camera sau");
            $('#eCameraAfter').focus();
            return;
        }
    } else if (eType === 'LapTop') {
        if (eIsDVD == '' || eIsDVD == 'undefined' || eIsDVD == null) {
            alert("Laptop có DVD không ?");
            $('#eIsDVD').focus();
            return;
        }
        if (eIsKeyBoardLight == '' || eIsKeyBoardLight == 'undefined' || eIsKeyBoardLight == null) {
            alert("Laptop có đèn bàn phím không");
            $('#eIsKeyBoardLight').focus();
            return;
        }
    }

    var electronic = {eType: eType, eName: eName, ePrice: ePrice, eQuantity: eQuantity,
        eCompanyID: eCompanyID, eScreenSize: eScreenSize, eOperaSystem: eOperaSystem, eWarrantyPeriod: eWarrantyPeriod
        , eBattery: eBattery, eCameraAfter: eCameraAfter, eIsDVD: eIsDVD, eIsKeyBoardLight: eIsKeyBoardLight
    }
    $.ajax({
        type: 'POST',
        url: '../AddElectronic',
        data: {eType: eType, eName: eName, ePrice: ePrice, eQuantity: eQuantity,
            eCompanyID: eCompanyID, eScreenSize: eScreenSize, eOperaSystem: eOperaSystem, eWarrantyPeriod: eWarrantyPeriod
            , eBattery: eBattery, eCameraAfter: eCameraAfter, eIsDVD: eIsDVD, eIsKeyBoardLight: eIsKeyBoardLight},
        dataType: 'html',
        success: function (result) {
            if (result == "ok") {
                alert("Thêm mới thành công");
                resetFormAddElectronic();
            } else {
                alert("Thêm mới thất bại");
//                $('#userName').focus();
            }
        },
        error: function (result) {
            alert("Lỗi Server");
        }
    });
});

function resetFormAddElectronic() {
    $('#eType').val('');
    $('#eName').val('');
    $('#ePrice').val('');
    $('#eQuantity').val('');
    $('#eCompanyID').val('');
    $('#eScreenSize').val('');
    $('#eOperaSystem').val('');
    $('#eWarrantyPeriod').val('');
    $('#gender').val('');
    $('#number').val('');
    resetAttr();
}

File: BOEC-finally-war/src/java/Admin/Controller/AddElectronic.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Admin.Controller;


import entities.Laptop;
import entities.cart.Category;
import entities.cart.Item;
import entities.item.electronic.Smartphone;
import entities.item.electronic.Tablet;
import java.io.IOException;
import java.io.PrintWriter;
import javax.ejb.EJB;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import session.cart.CategoryFacadeLocal;
import session.cart.ItemFacadeLocal;
import session.item.electronic.ElectronicFacadeLocal;
import session.item.electronic.LaptopFacadeLocal;
import session.item.electronic.SmartphoneFacadeLocal;
import session.item.electronic.TabletFacadeLocal;


/**
 *
 * @author HieuLe
 */
public class AddElectronic extends HttpServlet {
    
    @EJB
    private ItemFacadeLocal itemFacade;
    
    @EJB
    private ElectronicFacadeLocal electronicFacade;
    @EJB
    private SmartphoneFacadeLocal smartphoneFacade;
    @EJB
    private TabletFacadeLocal tabletFacade;
    @EJB
    private LaptopFacadeLocal laptopFacade;
    @EJB
    private CategoryFacadeLocal categoryFacade;
    
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/html;charset=UTF-8");
        try (PrintWriter out = response.getWriter()) {
            /* TODO output your page here. You may use following sample code. */
            out.println("<!DOCTYPE html>");
            out.println("<html>");
            out.println("<head>");
            out.println("<title>Servlet AddElectronic</title>");
            out.println("</head>");
            out.println("<body>");
            out.println("<h1>Servlet AddElectronic at " + request.getContextPath() + "</h1>");
            out.println("</body>");
            out.println("</html>");
        }
    }
    
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }
    
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        PrintWriter writer = response.getWriter();
        try {
            float eBattery = 0;
            String eCameraAfter = "";
            int eIsDVD;
            int eIsKeyBoardLight;
            String eType = request.getParameter("eType");
            String eName = request.getParameter("eName");
            int ePrice = Integer.parseInt(request.getParameter("ePrice"));
            int eQuantity = Integer.parseInt(request.getParameter("eQuantity"));
            String eCompanyID = request.getParameter("eCompanyID");
            float eScreenSize = Float.parseFloat(request.getParameter("eScreenSize"));
            String eOperaSystem = request.getParameter("eOperaSystem");
            float eWarrantyPeriod = Float.parseFloat(request.getParameter("eWarrantyPeriod"));
            
            Category category = categoryFacade.find("ELE"); // Danh mục ELectronic
            Item item = new Item("ITEM" + (1 + itemFacade.count()));
            item.setName(eName);
            item.setPrice(ePrice);
            item.setQuantity(eQuantity);
            item.setCategoryID(category);
//            item.setWarehouseEmployeePersonID(warehouseEmployeePersonID);
//            switch (){
//                case "SmartPhone": break;
//                case "Tablet": break;
//                case "LapTop": break;
//            }
            if (eType.equalsIgnoreCase("SmartPhone") || eType.equalsIgnoreCase("Tablet")) {       // Kiểm tra eType (Kiểu Electronic)
                eBattery = Float.parseFloat(request.getParameter("eBattery"));
                eCameraAfter = request.getParameter("eCameraAfter");
                if (eType.equalsIgnoreCase("SmartPhone")) {
                    Smartphone smartphone = new Smartphone();
                    smartphone.setItemID(item.getId());
                    smartphone.setItem(item);
                    smartphone.setBattery(eBattery);
                    smartphone.setCameraAfter(eCameraAfter);
                    smartphoneFacade.create(smartphone);
                } else if (eType.equalsIgnoreCase("Tablet")) {
                    Tablet tablet = new Tablet();
                    tablet.setItemID(item.getId());
                    tablet.setItem(item);
                    tablet.setBattery(eBattery);
                    tablet.setCameraAfter(eCameraAfter);
                    tabletFacade.create(tablet);
                }
            } else if (eType.equalsIgnoreCase("LapTop")) {
                eIsDVD = Integer.parseInt(request.getParameter("eIsDVD"));
                eIsKeyBoardLight = Integer.parseInt(request.getParameter("eIsKeyBoardLight"));
                Laptop laptop = new Laptop();
                laptop.setItemID(item.getId());
                laptop.setItem(item);
                laptop.setIsDVD(eIsDVD);
                laptop.setIsKeyBoardLight(eIsKeyBoardLight);
                laptopFacade.create(laptop);
            }
            // add Electronic
            writer.write("ok");
        } catch (Exception e) {
            e.printStackTrace();
            writer.write("fail");
        }
    }
    
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

}
